src/core/event_loop.rs

```diff
@@ -75,6 +75,19 @@ fn main_task<B: Backend + std::io::Write>(
         }
     };
 
+    // Check if there is a current song playing on startup
+    if let Some((_, song)) = ctx.find_current_song_in_queue() {
+        if let Some(command) = &ctx.config.on_song_change {
```

Comment:
Just a cosmetic nitpick since if let chains are finally stable in rust. Collapse the nested if let to a single block to reduce nesting.

```suggestion
    if let Some((_, song)) = ctx.find_current_song_in_queue()
        && let Some(command) = &ctx.config.on_song_change {
```